Hephaestus/Classes/Compiler.cs

```diff
@@ -0,0 +1,260 @@
+                            }
+                            else
+                            {
+                                // If the checksums are now different then we build the directory.
+                                BuildDirectory(sourceCodeDirectory, project);
+                            }
+                        }
+                        else
+                        {
+                            // If the JSON doesn't contain the current source code directories hash then we add it to the list
+                            //    but leave it as an empty hash for now.
+                            project.Hashes.Add(sourceCodeDirectory, new Hash());
+                            BuildDirectory(sourceCodeDirectory, project);
+                        }
+                    }
+                }
+
+                // Wait until all builders have closed.
+                while (LaunchedAddonBuilders + NotBuiltDirectories != SourceCodeDirectoryCount ||
+                       ExitedAddonBuilders + NotBuiltDirectories != SourceCodeDirectoryCount)
+                {
+                    Thread.Sleep(100);
+                }
+
+                // As it has finished building then we can safely launch the game (if configured).
+                if (project.StartGameAfterBuilding)
+                {
+                    // If none of the source code directories have changed, then ask the user if they want to start the game.
+                    // This prevents accidental launching of the game.
+                    if (NotBuiltDirectories == SourceCodeDirectoryCount)
+                    {
+                        if (ConsoleUtility.AskYesNoQuestion("Do you want to launch the game?"))
+                        {
+                            Game.Launch(project.Game);
+                        }
+                    }
+                    else
+                    {
+                        // If at least one source code directory did change then we launch anyways.
+                        Game.Launch(project.Game);
+                    }
+                }
+
+                // Wait until all builders have exited.
+                lock (OnBuilderExitLock)
+                {
+                    // Check if any non-zero exit codes are present in the ExitCodes list.
+                    // If it contains a non-zero exit code, then return an exit code of 1.
+                    int exitCode = ExitCodes.All(v => v == 0) ? 0 : 1;
+                    
+                    // If nothing has changed.
+                    if (NotBuiltDirectories == SourceCodeDirectoryCount)
+                    {
+                        // If the user does not want to rebuild then return the exit code (0, because nothing was built.)
+                        if (! ConsoleUtility.AskYesNoQuestion("Rebuild?"))
+                        {
+                            return 0;
+                        }
```

Comment:
I deliberately avoided this for the following reasons:
1. It can sometimes confuse readers
2. Code maintainability decreases as removing a simple space breaks the code
3. Codacy complains about not having the full style
4. Creating inconsistencies between different if statements.

Reply:
At least remove the `{}` in that case, you don't need those when there is only a single line in the if statement. And I have a distaste for it as it is more difficult to read.